docs/source/deprecation.md

```diff
@@ -1,14 +1,117 @@
+
+Since you marked the entity as deprecated, user will be notified via messages
+and reports when entity is used in the code. It is obvious that as an author
+of the deprecation, you know that the entity is deprecated and you do not want
+to notify people that the code - that still provides the deprecated
+functionality - is used. To suppress deprecation messages in such cases,
+use the `suppress_deprecation` decorator taking as arguments objects that
+should not be reported by the deprecation. E.g. in case you use it above the
+definition of an actor, any use of the deprecated entity inside the actor
+will not be reported.
+
+WARNING: It is forbidden to use the `suppress_deprecation` decorator for any
+purposes but the one, to keep original official functionality for a while.
+If you see the message and you are not *provider* of the functionality, you
+have to update your code to be independent on it.
+
+### Example of a model deprecation
+
+Imagine we want to deprecate a *Foo* model that is produced in an actor called
+*FooGenerator* and consumed in an actor called *FooConsumer*. Let's keep this
+example simple and say that we do not want to set any replacement of this
+model. First thing we have to do, is set the model definition as deprecated:
 
-To help with the deprecation process of entities in leapp, the leapp framework introduced a decorator `deprecated`
-which can be imported from `leapp.utils.deprecation`.
+```python
+from leapp.models import Model, fields
+from leapp.topics import SomeTopic
+from leapp.utils.deprecation import deprecated
+
+
+@deprecated(since='2020-06-20', message='This model has been deprecated.')
+class Foo(Model):
+    topic = SomeTopic
+    value = fields.String()
+```
+
+If we do this and execute actors that produce/consume messages of this model,
+we will immadiately obtain messages like these (just example from the actor
+producing the message after execution by snactor):
+
+```
+============================================================
+                 USE OF DEPRECATED ENTITIES
+============================================================
+
+Usage of deprecated Model "Foo" @ /path/to/repo/actors/foogenerator/actor.py:20
+Near:         self.produce(Foo)
+
+Reason: This model has been deprecated.
+------------------------------------------------------------
+
+============================================================
+             END OF USE OF DEPRECATED ENTITIES
+============================================================
+```
+
+Obviously we do not want to remove the actor yet, as in such a case, the
+model could be hardly called as deprecated. But we do not want the deprecation
+message to be produced in this case, as we want to keep the official
+functionality around that actor during the deprecation process, but the messages
+are focused on "custom" use of the model. E.g. in case when user create own
+actor producing/consuming a msg of the model. To fix this, suppress
```

Comment:
```suggestion
message to be produced in this case, as we want to keep the original
functionality of that actor during the deprecation protection period. The warning messages
are focused on a "custom" use of the deprecated model - in case a user creates their own
actor producing/consuming a message of the model. To fix this, suppress
```